libraries/chain/controller.cpp

```diff
@@ -1518,6 +1521,13 @@ struct controller_impl {
       auto& bb = pending->_block_stage.get<building_block>();
       const auto& pbhs = bb._pending_block_header_state;
 
+      const auto& gpo = self.get_global_properties();
+
+      const auto size = gpo.configuration.max_block_cpu_usage /
```

Comment:
This could potentially be very large. How about putting a cap on it, like 50000?

Reply:
Lets try a deque and see how that goes.

Reply:
`std::deque` is not nearly as good `boost::container::queue` configured to 1024 elements per block. However, the configurable boost queue is not available until boost 1.71. Going with std::queue for now as it is an improvement and we can switch to boost queue later.